From: Herbert Xu <herbert@gondor.apana.org.au>
Date: Wed, 27 Aug 2008 10:18:32 +1000
Subject: [CRYPTO] api: missing accessors for new crypto_alg field
Message-id: E1KY8k4-0002HC-00@gondolin.me.apana.org.au
O-Subject: [PATCH 10/19] [CRYPTO] api: Add missing accessors for new crypto_alg fields
Bugzilla: 446522

RHEL5 bugzilla #446522

[CRYPTO] api: Add missing accessors for new crypto_alg fields

Add missing accessors for cra_driver_name and cra_priority.

Signed-off-by: Michal Ludvig <michal@logix.cz>
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>

diff --git a/include/linux/crypto.h b/include/linux/crypto.h
index cda3885..88251bc 100644
--- a/include/linux/crypto.h
+++ b/include/linux/crypto.h
@@ -265,6 +265,16 @@ static inline const char *crypto_tfm_alg_name(struct crypto_tfm *tfm)
 	return tfm->__crt_alg->cra_name;
 }
 
+static inline const char *crypto_tfm_alg_driver_name(struct crypto_tfm *tfm)
+{
+	return tfm->__crt_alg->cra_driver_name;
+}
+
+static inline int crypto_tfm_alg_priority(struct crypto_tfm *tfm)
+{
+	return tfm->__crt_alg->cra_priority;
+}
+
 static inline const char *crypto_tfm_alg_modname(struct crypto_tfm *tfm)
 {
 	return module_name(tfm->__crt_alg->cra_module);
diff --git a/include/linux/ncrypto.h b/include/linux/ncrypto.h
index 63169ff..d76d548 100644
--- a/include/linux/ncrypto.h
+++ b/include/linux/ncrypto.h
@@ -415,6 +415,16 @@ static inline const char *ncrypto_tfm_alg_name(struct ncrypto_tfm *tfm)
 	return tfm->__crt_alg->cra_name;
 }
 
+static inline const char *ncrypto_tfm_alg_driver_name(struct ncrypto_tfm *tfm)
+{
+	return tfm->__crt_alg->cra_driver_name;
+}
+
+static inline int ncrypto_tfm_alg_priority(struct ncrypto_tfm *tfm)
+{
+	return tfm->__crt_alg->cra_priority;
+}
+
 static inline u32 ncrypto_tfm_alg_type(struct ncrypto_tfm *tfm)
 {
 	return tfm->__crt_alg->cra_flags & NCRYPTO_ALG_TYPE_MASK;
